src/f32-rwsum/f32-rwdsum-1p1x-scalar-c1.c

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 Google LLC
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include <assert.h>
+
+#include "xnnpack/common.h"
+#include "xnnpack/reduce.h"
+
+#define CEILING_POS(X) ((X-(int)(X)) > 0 ? (int)(X+1) : (int)(X))
+#define CEILING_NEG(X) (int)(X)
+#define CEIL(X) ( ((X) > 0) ? CEILING_POS(X) : CEILING_NEG(X) )
+#define FLOORING_POS(X) (int)(X)
+#define FLOORING_NEG(X) ((X-(int)(X)) > 0 ? (int)(X-1) : (int)(X))
+#define FLOOR(X) ( ((X) > 0) ? FLOORING_POS(X) : FLOORING_NEG(X) )
+#define MAX(X,Y) (X > Y ? X : Y)
+#define MIN(X,Y) ((X) < (Y) ? (X) : (Y))
+
+// reduce window for vertical reduction 
+void xnn_f32_rwdsum_ukernel_1p1x__scalar_c1(
+    size_t rows,
+    size_t channels,
+    const float* input,
+    float init_value,
+    int* padding, 
+    int base_dilation, 
```

Comment:
Do these need to be `int`, i.e. signed? If we can assume they are positive, a lot of the necessary arithmetic gets simpler (e.g. `divide_round_up` is pretty simple for `size_t`, not so much for `int`).